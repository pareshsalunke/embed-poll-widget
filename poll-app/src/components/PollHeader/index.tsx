import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { Typography } from '@material-ui/core';

interface Props {
  color?: string;
  children?: React.ReactNode;
}

const useStyles = makeStyles( {
  panelHeader: {
    backgroundColor: '#2980b9',
    color: '#fff',
    padding: '20px',
    margin: 0,

    '&.MuiTypography-h4': {
      fontSize: '1.75rem'
    }
  },

});

const PollHeader = (props: Props) => {
  const {
    children
  } = props;
  const classes = useStyles();
  return (
      <Typography variant="h4" className={classes.panelHeader}>{children}</Typography>
  );
}

export default PollHeader;