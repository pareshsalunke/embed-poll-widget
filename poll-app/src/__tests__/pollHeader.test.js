import React from 'react';
import PollHeader from '../components/PollHeader';
import { render } from '@testing-library/react';

describe('Poll Header component', () => {
  let Comp;
  beforeEach(() => {
    Comp = () => {

      return (
        <PollHeader>Header</PollHeader>
      );
    }
  });

  test('error component mounts', () => {
    expect(Comp()).toMatchSnapshot();
  });

  test('error component text', () => {
    const { getByText } = render(<Comp />);
    expect(getByText("Header")).toBeInTheDocument();
  });
})