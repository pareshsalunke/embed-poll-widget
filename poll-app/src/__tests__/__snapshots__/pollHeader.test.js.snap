// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Poll Header component error component mounts 1`] = `
<PollHeader>
  Header
</PollHeader>
`;
